[100%] Testing the TCP connection...
Test project /mnt/i/DIY/CS144/sponge/build
        Start   1: t_wrapping_ints_cmp
  1/159 Test   #1: t_wrapping_ints_cmp ..............   Passed    0.01 sec
        Start   2: t_wrapping_ints_unwrap
  2/159 Test   #2: t_wrapping_ints_unwrap ...........   Passed    0.02 sec
        Start   3: t_wrapping_ints_wrap
  3/159 Test   #3: t_wrapping_ints_wrap .............   Passed    0.02 sec
        Start   4: t_recv_connect
  4/159 Test   #4: t_recv_connect ...................   Passed    0.05 sec
        Start   5: t_recv_transmit
  5/159 Test   #5: t_recv_transmit ..................   Passed    0.31 sec
        Start   6: t_recv_window
  6/159 Test   #6: t_recv_window ....................   Passed    0.04 sec
        Start   7: t_recv_reorder
  7/159 Test   #7: t_recv_reorder ...................   Passed    0.04 sec
        Start   8: t_recv_close
  8/159 Test   #8: t_recv_close .....................   Passed    0.04 sec
        Start   9: t_send_connect
  9/159 Test   #9: t_send_connect ...................   Passed    0.04 sec
        Start  10: t_send_transmit
 10/159 Test  #10: t_send_transmit ..................   Passed    0.29 sec
        Start  11: t_send_retx
 11/159 Test  #11: t_send_retx ......................   Passed    0.05 sec
        Start  12: t_send_window
 12/159 Test  #12: t_send_window ....................   Passed    0.20 sec
        Start  13: t_send_ack
 13/159 Test  #13: t_send_ack .......................   Passed    0.05 sec
        Start  14: t_send_close
 14/159 Test  #14: t_send_close .....................   Passed    0.05 sec
        Start  15: t_strm_reassem_single
 15/159 Test  #15: t_strm_reassem_single ............   Passed    0.02 sec
        Start  16: t_strm_reassem_seq
 16/159 Test  #16: t_strm_reassem_seq ...............   Passed    0.02 sec
        Start  17: t_strm_reassem_dup
 17/159 Test  #17: t_strm_reassem_dup ...............   Passed    0.03 sec
        Start  18: t_strm_reassem_holes
 18/159 Test  #18: t_strm_reassem_holes .............   Passed    0.02 sec
        Start  19: t_strm_reassem_many
 19/159 Test  #19: t_strm_reassem_many ..............   Passed    8.55 sec
        Start  20: t_strm_reassem_overlapping
 20/159 Test  #20: t_strm_reassem_overlapping .......   Passed    0.03 sec
        Start  21: t_strm_reassem_win
 21/159 Test  #21: t_strm_reassem_win ...............   Passed    8.89 sec
        Start  22: t_strm_reassem_cap
 22/159 Test  #22: t_strm_reassem_cap ...............   Passed    0.76 sec
        Start  23: t_byte_stream_construction
 23/159 Test  #23: t_byte_stream_construction .......   Passed    0.02 sec
        Start  24: t_byte_stream_one_write
 24/159 Test  #24: t_byte_stream_one_write ..........   Passed    0.02 sec
        Start  25: t_byte_stream_two_writes
 25/159 Test  #25: t_byte_stream_two_writes .........   Passed    0.02 sec
        Start  26: t_byte_stream_capacity
 26/159 Test  #26: t_byte_stream_capacity ...........   Passed    4.32 sec
        Start  27: t_byte_stream_many_writes
 27/159 Test  #27: t_byte_stream_many_writes ........   Passed    0.05 sec
        Start  28: t_webget
 28/159 Test  #28: t_webget .........................   Passed    1.87 sec
        Start  31: t_tcp_parser
 29/159 Test  #31: t_tcp_parser .....................   Passed    0.05 sec
        Start  32: t_ipv4_parser
 30/159 Test  #32: t_ipv4_parser ....................   Passed    0.06 sec
        Start  33: t_active_close
 31/159 Test  #33: t_active_close ...................***Failed    0.10 sec
Test Failure on expectation:
	Expectation: exactly one segment sent with (A=1,ackno=2,)

Failure message:
	The TCP an extra segment when it should not have

List of steps that executed successfully:
	Action:      connect
	Expectation: exactly one segment sent with (A=0,R=0,S=1,F=0,seqno=0,payload_size=0,)
	Action:      packet arrives: Header(flags=SA,seqno=0,ack=1,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=1,payload_size=0,)
	Action:      close
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=1,ackno=1,seqno=1,)
	Action:      packet arrives: Header(flags=A,seqno=1,ack=2,win=137) with no payload
	Action:      4000ms pass
	Expectation: TCP in state sender=`stream finished and fully acknowledged`, receiver=`SYN received (ackno exists), and input to stream hasn't ended`, active=1, linger_after_streams_finish=1
	Action:      packet arrives: Header(flags=AF,seqno=1,ack=2,win=137) with no payload
	Action:      1ms pass

Note:
	test 3 failed: wrong ACK for FIN

Warning: Unclean shutdown of TCPConnection
The TCP an extra segment when it should not have

        Start  34: t_passive_close
 32/159 Test  #34: t_passive_close ..................***Failed    0.05 sec
Test Failure on expectation:
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=2,)

Failure message:
	The TCP should have produced a segment that existed, but it did not

List of steps that executed successfully:
	Action:      connect
	Expectation: exactly one segment sent with (A=0,R=0,S=1,F=0,seqno=0,payload_size=0,)
	Action:      packet arrives: Header(flags=SA,seqno=0,ack=1,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=1,payload_size=0,)
	Action:      packet arrives: Header(flags=AF,seqno=1,ack=1,win=137) with no payload

Warning: Unclean shutdown of TCPConnection
The TCP should have produced a segment that existed, but it did not

        Start  36: t_ack_rst
 33/159 Test  #36: t_ack_rst ........................***Failed    0.05 sec
Test 1
Test Failure on expectation:
	Expectation: TCP has 0 unassembled bytes

Failure message:
	The TCP has `unassembled_bytes = 1`, but unassembled_bytes was expected to be `0`

List of steps that executed successfully:
	Action:      connect
	Expectation: exactly one segment sent with (A=0,R=0,S=1,F=0,seqno=2147483647,payload_size=0,)
	Action:      packet arrives: Header(flags=SA,seqno=2147483647,ack=2147483648,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=2147483648,payload_size=0,)
	Action:      packet arrives: Header(flags=A,seqno=2147483648,ack=2147483648,win=137) with no payload
	Expectation: no (more) segments sent
	Action:      packet arrives: Header(flags=A,seqno=2147483648,ack=2147483647,win=137) with no payload
	Expectation: no (more) segments sent
	Action:      packet arrives: Header(flags=A,seqno=2147483647,ack=2147483648,win=137) with 1 data bytes "<1>"

Note:
	test 1 failed: seg queued on early seqno

Warning: Unclean shutdown of TCPConnection
The TCP has `unassembled_bytes = 1`, but unassembled_bytes was expected to be `0`

        Start  38: t_ack_rst_win
 34/159 Test  #38: t_ack_rst_win ....................***Failed    0.06 sec
Test Failure on expectation:
	Expectation: TCP has 0 unassembled bytes

Failure message:
	The TCP has `unassembled_bytes = 1`, but unassembled_bytes was expected to be `0`

List of steps that executed successfully:
	Action:      connect
	Expectation: exactly one segment sent with (A=0,R=0,S=1,F=0,seqno=2147483647,payload_size=0,)
	Action:      packet arrives: Header(flags=SA,seqno=2147483647,ack=2147483648,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=2147483648,payload_size=0,)
	Action:      packet arrives: Header(flags=A,seqno=2147483648,ack=2147483648,win=137) with no payload
	Expectation: no (more) segments sent
	Action:      packet arrives: Header(flags=A,seqno=2147483648,ack=2147483647,win=137) with no payload
	Expectation: no (more) segments sent
	Action:      packet arrives: Header(flags=A,seqno=2147483647,ack=2147483648,win=137) with 1 data bytes "<1>"

Note:
	test 1 failed: seg queued on early seqno

Warning: Unclean shutdown of TCPConnection
The TCP has `unassembled_bytes = 1`, but unassembled_bytes was expected to be `0`

        Start  40: t_connect
 35/159 Test  #40: t_connect ........................   Passed    0.04 sec
        Start  42: t_listen
 36/159 Test  #42: t_listen .........................   Passed    0.06 sec
        Start  43: t_winsize
 37/159 Test  #43: t_winsize ........................***Failed    0.05 sec
Test Failure on expectation:
	Action:      write (31760 bytes with 31760 accepted) [9f, b9, 35, fc, 1d, 16, 6, 20, d2, 6b, 35, a9, 1b, 4f, fa, 2b, ...]

Failure message:
	31760 bytes should have been written but 0 were

List of steps that executed successfully:
	Action:      listen
	Action:      packet arrives: Header(flags=S,seqno=680192104,ack=0,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,ackno=680192105,win=5761,)
	Action:      packet arrives: Header(flags=A,seqno=680192105,ack=2950747025,win=15880) with no payload
	Expectation: no (more) segments sent
	Expectation: TCP in state sender=`stream ongoing`, receiver=`SYN received (ackno exists), and input to stream hasn't ended`, active=1, linger_after_streams_finish=1

Warning: Unclean shutdown of TCPConnection
31760 bytes should have been written but 0 were

        Start  45: t_retx
 38/159 Test  #45: t_retx ...........................***Failed    0.06 sec
  check_segment
Test Failure on expectation:
	Expectation: segment sent with (A=1,payload_size=4,data="asdf",)

Failure message:
	The TCP should have produced a segment that existed, but it did not

List of steps that executed successfully:
	Action:      connect
	Expectation: exactly one segment sent with (A=0,R=0,S=1,F=0,seqno=2959204975,payload_size=0,)
	Action:      packet arrives: Header(flags=SA,seqno=2959204975,ack=2959204976,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=2959204976,payload_size=0,)
	Action:      write (4 bytes) [61, 73, 64, 66]
	Action:      1ms pass

Warning: Unclean shutdown of TCPConnection
The TCP should have produced a segment that existed, but it did not (in check_segment called from line 36)

        Start  46: t_retx_win
 39/159 Test  #46: t_retx_win .......................***Failed    0.04 sec
Test Failure on expectation:
	Expectation: segment sent

Failure message:
	The TCP should have produces a segment, but did not

List of steps that executed successfully:
	Action:      connect
	Expectation: exactly one segment sent with (A=0,R=0,S=1,F=0,seqno=480530959,payload_size=0,)
	Action:      packet arrives: Header(flags=SA,seqno=480530959,ack=480530960,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=480530960,payload_size=0,)
	Action:      write (4 bytes) [61, 73, 64, 66]
	Action:      1ms pass
	Action:      20ms pass
	Action:      write (4 bytes) [71, 77, 65, 72]
	Action:      1ms pass

Note:
	test 2 failed: cannot read after write()s

Warning: Unclean shutdown of TCPConnection
The TCP should have produces a segment, but did not

        Start  47: t_loopback
 40/159 Test  #47: t_loopback .......................***Failed    0.05 sec
Test Failure on expectation:
	Expectation: TCP has 4308 bytes in flight

Failure message:
	The TCP has `bytes_in_flight = 0`, but bytes_in_flight was expected to be `4308`

List of steps that executed successfully:
	Action:      connect
	Expectation: exactly one segment sent with (A=0,R=0,S=1,F=0,seqno=2192219848,payload_size=0,)
	Action:      packet arrives: Header(flags=SA,seqno=2192219848,ack=2192219849,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=2192219849,payload_size=0,)
	Action:      packet arrives: Header(flags=A,seqno=2192219849,ack=2192219849,win=65000) with no payload
	Action:      write (4308 bytes) [a6, be, 26, 16, 4c, 5a, 10, f6, e, 21, d7, a2, ea, b6, 26, 9e, ...]
	Action:      1ms pass

Warning: Unclean shutdown of TCPConnection
The TCP has `bytes_in_flight = 0`, but bytes_in_flight was expected to be `4308`

        Start  48: t_loopback_win
 41/159 Test  #48: t_loopback_win ...................***Failed    0.05 sec
Test Failure on expectation:
	Expectation: segment sent

Failure message:
	The TCP should have produces a segment, but did not

List of steps that executed successfully:
	Action:      connect
	Expectation: exactly one segment sent with (A=0,R=0,S=1,F=0,seqno=4221583034,payload_size=0,)
	Action:      packet arrives: Header(flags=SA,seqno=4221583034,ack=4221583035,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=4221583035,payload_size=0,)
	Action:      packet arrives: Header(flags=A,seqno=4221583035,ack=4221583035,win=65000) with no payload
	Action:      write (3953 bytes) [94, bc, 95, ed, 46, de, 9e, ba, 19, d3, f8, d3, 64, d9, a1, 5, ...]
	Action:      1ms pass

Note:
	test 2 failed: cannot read after write()

Warning: Unclean shutdown of TCPConnection
The TCP should have produces a segment, but did not

        Start  49: t_reorder
 42/159 Test  #49: t_reorder ........................***Failed    0.07 sec
Test Failure on expectation:
	segment sent with (A=1,)
Failure message:
	The TCP should have produced a segment that existed, but it did not

List of steps that executed successfully:
	Action:      connect
	Expectation: exactly one segment sent with (A=0,R=0,S=1,F=0,seqno=3577302327,payload_size=0,)
	Action:      packet arrives: Header(flags=SA,seqno=1108946742,ack=3577302328,win=137) with no payload
	Expectation: exactly one segment sent with (A=1,R=0,S=0,F=0,ackno=1108946743,payload_size=0,)
	Action:      packet arrives: Header(flags=A,seqno=1108953133,ack=3577302328,win=137) with 224 data bytes "<238>y<12>j<198><242><226><200><254><19>_h<169>u-<241>..."

Note:
	test 1 failed: no ACK for datagram

Warning: Unclean shutdown of TCPConnection
The TCP should have produced a segment that existed, but it did not

        Start  50: t_address_dt
 43/159 Test  #50: t_address_dt .....................   Passed    0.02 sec
        Start  51: t_parser_dt
 44/159 Test  #51: t_parser_dt ......................   Passed    0.02 sec
        Start  52: t_socket_dt
 45/159 Test  #52: t_socket_dt ......................   Passed    0.03 sec
        Start  53: t_udp_client_send
 46/159 Test  #53: t_udp_client_send ................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:15463... done.
new connection from 169.254.144.1:53408.
DEBUG: Outbound stream to 169.254.144.1:53408 finished (1 byte still in flight).
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14283 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  54: t_udp_server_send
 47/159 Test  #54: t_udp_server_send ................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:4652... done.
new connection from 169.254.144.1:41805.
DEBUG: Outbound stream to 169.254.144.1:4652 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:41805 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  55: t_udp_client_recv
 48/159 Test  #55: t_udp_client_recv ................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:19948... done.
new connection from 169.254.144.1:35181.
DEBUG: Outbound stream to 169.254.144.1:19948 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:35181 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  56: t_udp_server_recv
 49/159 Test  #56: t_udp_server_recv ................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:3069... done.
new connection from 169.254.144.1:35316.
DEBUG: Outbound stream to 169.254.144.1:35316 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:3069 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14399 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  57: t_udp_client_dupl
 50/159 Test  #57: t_udp_client_dupl ................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:2414... done.
new connection from 169.254.144.1:46020.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14418 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  58: t_udp_server_dupl
 51/159 Test  #58: t_udp_server_dupl ................***Failed    0.18 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:4607... done.
new connection from 169.254.144.1:52941.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14435 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14437 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  59: t_ucS_1M_32k
 52/159 Test  #59: t_ucS_1M_32k .....................***Failed    0.24 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:14390... done.
new connection from 169.254.144.1:38712.
DEBUG: Outbound stream to 169.254.144.1:38712 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:14390 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14457 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  60: t_ucS_128K_8K
 53/159 Test  #60: t_ucS_128K_8K ....................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:32965... done.
new connection from 169.254.144.1:53223.
DEBUG: Outbound stream to 169.254.144.1:53223 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:32965 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14477 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  61: t_ucS_16_1
 54/159 Test  #61: t_ucS_16_1 .......................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:28140... done.
new connection from 169.254.144.1:34368.
DEBUG: Outbound stream to 169.254.144.1:34368 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:28140 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14497 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  62: t_ucS_32K_d
 55/159 Test  #62: t_ucS_32K_d ......................***Failed    0.18 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:13334... done.
new connection from 169.254.144.1:39008.
DEBUG: Outbound stream to 169.254.144.1:39008 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:13334 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14518 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  63: t_ucR_1M_32k
 56/159 Test  #63: t_ucR_1M_32k .....................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:33486... done.
new connection from 169.254.144.1:59812.
DEBUG: Outbound stream to 169.254.144.1:33486 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:59812 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  64: t_ucR_128K_8K
 57/159 Test  #64: t_ucR_128K_8K ....................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:29645... done.
new connection from 169.254.144.1:59392.
DEBUG: Outbound stream to 169.254.144.1:29645 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:59392 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  65: t_ucR_16_1
 58/159 Test  #65: t_ucR_16_1 .......................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:17030... done.
new connection from 169.254.144.1:37656.
DEBUG: Outbound stream to 169.254.144.1:17030 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:37656 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  66: t_ucR_32K_d
 59/159 Test  #66: t_ucR_32K_d ......................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:22846... done.
new connection from 169.254.144.1:38427.
DEBUG: Outbound stream to 169.254.144.1:22846 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:38427 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  67: t_ucD_1M_32k
 60/159 Test  #67: t_ucD_1M_32k .....................***Failed    0.25 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:29821... done.
new connection from 169.254.144.1:35243.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14681 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  68: t_ucD_128K_8K
 61/159 Test  #68: t_ucD_128K_8K ....................***Failed    0.23 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:5997... done.
new connection from 169.254.144.1:34829.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14700 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14702 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  69: t_ucD_16_1
 62/159 Test  #69: t_ucD_16_1 .......................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:15084... done.
new connection from 169.254.144.1:58054.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14721 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14723 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  70: t_ucD_32K_d
 63/159 Test  #70: t_ucD_32K_d ......................***Failed    0.18 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:3907... done.
new connection from 169.254.144.1:37041.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14740 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14742 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  71: t_usS_1M_32k
 64/159 Test  #71: t_usS_1M_32k .....................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:2318... done.
new connection from 169.254.144.1:41256.
DEBUG: Outbound stream to 169.254.144.1:2318 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:41256 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  72: t_usS_128K_8K
 65/159 Test  #72: t_usS_128K_8K ....................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:4494... done.
new connection from 169.254.144.1:44294.
DEBUG: Outbound stream to 169.254.144.1:4494 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:44294 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  73: t_usS_16_1
 66/159 Test  #73: t_usS_16_1 .......................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:26711... done.
new connection from 169.254.144.1:46239.
DEBUG: Outbound stream to 169.254.144.1:26711 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:46239 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  74: t_usS_32K_d
 67/159 Test  #74: t_usS_32K_d ......................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:15000... done.
new connection from 169.254.144.1:39437.
DEBUG: Outbound stream to 169.254.144.1:15000 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:39437 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  75: t_usR_1M_32k
 68/159 Test  #75: t_usR_1M_32k .....................***Failed    0.27 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:27870... done.
new connection from 169.254.144.1:56399.
DEBUG: Outbound stream to 169.254.144.1:56399 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:27870 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14902 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  76: t_usR_128K_8K
 69/159 Test  #76: t_usR_128K_8K ....................***Failed    0.21 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:8492... done.
new connection from 169.254.144.1:56726.
DEBUG: Outbound stream to 169.254.144.1:56726 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:8492 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14949 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  77: t_usR_16_1
 70/159 Test  #77: t_usR_16_1 .......................***Failed    0.16 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:31060... done.
new connection from 169.254.144.1:41760.
DEBUG: Outbound stream to 169.254.144.1:41760 finished (1 byte still in flight).
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14980 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  78: t_usR_32K_d
 71/159 Test  #78: t_usR_32K_d ......................***Failed    0.17 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:30954... done.
new connection from 169.254.144.1:45812.
DEBUG: Outbound stream to 169.254.144.1:45812 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:30954 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 14998 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  79: t_usD_1M_32k
 72/159 Test  #79: t_usD_1M_32k .....................***Failed    0.28 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:6918... done.
new connection from 169.254.144.1:47102.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15017 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15019 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  80: t_usD_128K_8K
 73/159 Test  #80: t_usD_128K_8K ....................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:30104... done.
new connection from 169.254.144.1:45376.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15049 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  81: t_usD_16_1
 74/159 Test  #81: t_usD_16_1 .......................***Failed    0.17 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:15758... done.
new connection from 169.254.144.1:33416.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15068 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15070 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  82: t_usD_32K_d
 75/159 Test  #82: t_usD_32K_d ......................***Failed    0.17 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:7582... done.
new connection from 169.254.144.1:40813.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15089 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  83: t_ucS_128K_8K_l
 76/159 Test  #83: t_ucS_128K_8K_l ..................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:9067... done.
new connection from 169.254.144.1:59385.
DEBUG: Outbound stream to 169.254.144.1:59385 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:9067 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15109 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  84: t_ucS_128K_8K_L
 77/159 Test  #84: t_ucS_128K_8K_L ..................***Failed    0.20 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:2889... done.
new connection from 169.254.144.1:35789.
DEBUG: Outbound stream to 169.254.144.1:35789 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:2889 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15132 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  85: t_ucS_128K_8K_lL
 78/159 Test  #85: t_ucS_128K_8K_lL .................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:4981... 

        Start  86: t_ucR_128K_8K_l
 79/159 Test  #86: t_ucR_128K_8K_l ..................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:16836... done.
new connection from 169.254.144.1:33082.
DEBUG: Outbound stream to 169.254.144.1:16836 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:33082 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  87: t_ucR_128K_8K_L
 80/159 Test  #87: t_ucR_128K_8K_L ..................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:28184... done.
new connection from 169.254.144.1:49898.
DEBUG: Outbound stream to 169.254.144.1:28184 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:49898 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  88: t_ucR_128K_8K_lL
 81/159 Test  #88: t_ucR_128K_8K_lL .................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:14682... done.
new connection from 169.254.144.1:43881.
DEBUG: Outbound stream to 169.254.144.1:14682 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:43881 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  89: t_ucD_128K_8K_l
 82/159 Test  #89: t_ucD_128K_8K_l ..................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:17063... 

        Start  90: t_ucD_128K_8K_L
 83/159 Test  #90: t_ucD_128K_8K_L ..................***Failed    0.24 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:5249... done.
new connection from 169.254.144.1:55551.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15391 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  91: t_ucD_128K_8K_lL
 84/159 Test  #91: t_ucD_128K_8K_lL .................***Failed    0.27 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:24765... done.
new connection from 169.254.144.1:47406.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15415 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
Error in _rt_connect
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 156: 15413 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"

        Start  92: t_usS_128K_8K_l
 85/159 Test  #92: t_usS_128K_8K_l ..................***Timeout  10.02 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:25833... done.
DEBUG: Outbound stream to 169.254.144.1:25833 finished (1 byte still in flight).
new connection from 169.254.144.1:52914.
DEBUG: Inbound stream from 169.254.144.1:52914 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  93: t_usS_128K_8K_L
 86/159 Test  #93: t_usS_128K_8K_L ..................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:18185... done.
new connection from 169.254.144.1:58443.
DEBUG: Outbound stream to 169.254.144.1:18185 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:58443 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  94: t_usS_128K_8K_lL
 87/159 Test  #94: t_usS_128K_8K_lL .................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:9283... done.
DEBUG: Outbound stream to 169.254.144.1:9283 finished (1 byte still in flight).
new connection from 169.254.144.1:33626.
DEBUG: Inbound stream from 169.254.144.1:33626 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start  95: t_usR_128K_8K_l
 88/159 Test  #95: t_usR_128K_8K_l ..................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:18927... 

        Start  96: t_usR_128K_8K_L
 89/159 Test  #96: t_usR_128K_8K_L ..................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:7704... 

        Start  97: t_usR_128K_8K_lL
 90/159 Test  #97: t_usR_128K_8K_lL .................***Failed    1.80 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:5040... done.
new connection from 169.254.144.1:34141.
DEBUG: Outbound stream to 169.254.144.1:34141 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.1:5040 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15710 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  98: t_usD_128K_8K_l
 91/159 Test  #98: t_usD_128K_8K_l ..................***Failed    0.27 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:2854... done.
new connection from 169.254.144.1:57377.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15730 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15734 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start  99: t_usD_128K_8K_L
 92/159 Test  #99: t_usD_128K_8K_L ..................***Failed    0.22 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:2108... done.
new connection from 169.254.144.1:48345.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15755 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 100: t_usD_128K_8K_lL
 93/159 Test #100: t_usD_128K_8K_lL .................***Failed    0.28 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.1:18478... done.
new connection from 169.254.144.1:34455.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15774 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15778 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 101: t_ipv4_client_send
 94/159 Test #101: t_ipv4_client_send ...............***Failed    0.20 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:7290... done.
new connection from 169.254.144.9:26318.
DEBUG: Outbound stream to 169.254.144.9:26318 finished (1 byte still in flight).
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15799 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 102: t_ipv4_server_send
 95/159 Test #102: t_ipv4_server_send ...............***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:14149... done.
DEBUG: Outbound stream to 169.254.144.9:14149 finished (1 byte still in flight).
new connection from 169.254.145.9:35512.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 103: t_ipv4_client_recv
 96/159 Test #103: t_ipv4_client_recv ...............***Timeout  10.01 sec
DEBUG: Connecting to DEBUG: Listening for incoming connection... 169.254.145.9:32946... done.
DEBUG: Outbound stream to 169.254.145.9:32946 finished (1 byte still in flight).
new connection from 169.254.144.9:32090.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 104: t_ipv4_server_recv
 97/159 Test #104: t_ipv4_server_recv ...............***Failed    0.18 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:25527... done.
new connection from 169.254.145.9:5559.
DEBUG: Outbound stream to 169.254.145.9:5559 finished (1 byte still in flight).
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15910 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 105: t_ipv4_client_dupl
 98/159 Test #105: t_ipv4_client_dupl ...............***Failed    0.17 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:14767... done.
new connection from 169.254.144.9:6345.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15927 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15929 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 106: t_ipv4_server_dupl
 99/159 Test #106: t_ipv4_server_dupl ...............***Failed    0.17 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:17860... done.
new connection from 169.254.145.9:16493.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15946 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15948 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 107: t_icS_1M_32k
100/159 Test #107: t_icS_1M_32k .....................***Failed    0.23 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:20689... done.
new connection from 169.254.144.9:13486.
DEBUG: Outbound stream to 169.254.144.9:13486 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.145.9:20689 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15968 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 108: t_icS_128K_8K
101/159 Test #108: t_icS_128K_8K ....................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:12099... done.
new connection from 169.254.144.9:7029.
DEBUG: Outbound stream to 169.254.144.9:7029 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.145.9:12099 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 15988 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 109: t_icS_16_1
102/159 Test #109: t_icS_16_1 .......................***Failed    0.21 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:19128... done.
new connection from 169.254.144.9:6545.
DEBUG: Outbound stream to 169.254.144.9:6545 finished (1 byte still in flight).
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16008 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 110: t_icS_32K_d
103/159 Test #110: t_icS_32K_d ......................***Failed    0.17 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:10381... done.
new connection from 169.254.144.9:240.
DEBUG: Outbound stream to 169.254.144.9:240 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.145.9:10381 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16026 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 111: t_icR_1M_32k
104/159 Test #111: t_icR_1M_32k .....................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:7756... done.
DEBUG: Outbound stream to 169.254.145.9:7756 finished (1 byte still in flight).
new connection from 169.254.144.9:41896.
DEBUG: Inbound stream from 169.254.144.9:41896 finished cleanly.
DEBUG: Outbound stream to 169.254.145.9:7756 has been fully acknowledged.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 112: t_icR_128K_8K
105/159 Test #112: t_icR_128K_8K ....................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:15468... done.
DEBUG: Outbound stream to 169.254.145.9:15468 finished (1 byte still in flight).
new connection from 169.254.144.9:3990.
DEBUG: Inbound stream from 169.254.144.9:3990 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 113: t_icR_16_1
106/159 Test #113: t_icR_16_1 .......................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:12636... done.
DEBUG: Outbound stream to new connection from 169.254.144.9:63929.
169.254.145.9:12636 finished (1 byte still in flight).
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 114: t_icR_32K_d
107/159 Test #114: t_icR_32K_d ......................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:2301... done.
DEBUG: Outbound stream to 169.254.145.9:2301 finished (1 byte still in flight).
new connection from 169.254.144.9:21411.
DEBUG: Inbound stream from 169.254.144.9:21411 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 115: t_icD_1M_32k
108/159 Test #115: t_icD_1M_32k .....................***Failed    0.49 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:33566... done.
new connection from 169.254.144.9:2016.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16264 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 116: t_icD_128K_8K
109/159 Test #116: t_icD_128K_8K ....................***Failed    0.31 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:31693... done.
new connection from 169.254.144.9:58340.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16285 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16288 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 117: t_icD_16_1
110/159 Test #117: t_icD_16_1 .......................***Failed    0.25 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:29278... done.
new connection from 169.254.144.9:48170.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16307 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16309 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 118: t_icD_32K_d
111/159 Test #118: t_icD_32K_d ......................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:10370... done.
new connection from 169.254.144.9:48543.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16329 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 119: t_isS_1M_32k
112/159 Test #119: t_isS_1M_32k .....................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:16418... done.
DEBUG: Outbound stream to 169.254.144.9:16418 finished (1 byte still in flight).
new connection from 169.254.145.9:22701.
DEBUG: Inbound stream from 169.254.145.9:22701 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 120: t_isS_128K_8K
113/159 Test #120: t_isS_128K_8K ....................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:5189... done.
DEBUG: Outbound stream to 169.254.144.9:5189 finished (1 byte still in flight).
new connection from 169.254.145.9:43653.
DEBUG: Inbound stream from 169.254.145.9:43653 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 121: t_isS_16_1
114/159 Test #121: t_isS_16_1 .......................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:31215... done.
DEBUG: Outbound stream to new connection from 169.254.145.9:58152.
169.254.144.9:31215 finished (1 byte still in flight).
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 122: t_isS_32K_d
115/159 Test #122: t_isS_32K_d ......................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:27261... done.
new connection from DEBUG: Outbound stream to 169.254.145.9:35087.
169.254.144.9:27261 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.145.9:35087 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 123: t_isR_1M_32k
116/159 Test #123: t_isR_1M_32k .....................***Failed    0.28 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:20600... done.
new connection from 169.254.145.9:42937.
DEBUG: Outbound stream to 169.254.145.9:42937 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.9:20600 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16579 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 124: t_isR_128K_8K
117/159 Test #124: t_isR_128K_8K ....................***Failed    0.65 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:2757... done.
new connection from 169.254.145.9:28638.
DEBUG: Outbound stream to 169.254.145.9:28638 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.9:2757 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16602 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 125: t_isR_16_1
118/159 Test #125: t_isR_16_1 .......................***Failed    0.17 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:26906... done.
new connection from 169.254.145.9:22633.
DEBUG: Outbound stream to 169.254.145.9:22633 finished (1 byte still in flight).
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16622 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 126: t_isR_32K_d
119/159 Test #126: t_isR_32K_d ......................***Failed    0.17 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:29490... done.
new connection from 169.254.145.9:9938.
DEBUG: Outbound stream to 169.254.145.9:9938 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.9:29490 finished cleanly.
DEBUG: Waiting for clean shutdown... Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16640 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 127: t_isD_1M_32k
120/159 Test #127: t_isD_1M_32k .....................***Failed    0.28 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:14070... done.
new connection from 169.254.145.9:7316.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16670 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect
/mnt/i/DIY/CS144/sponge/txrx.sh: line 156: 16668 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"

        Start 128: t_isD_128K_8K
121/159 Test #128: t_isD_128K_8K ....................***Failed    2.32 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:10956... done.
new connection from 169.254.145.9:52069.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16697 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 129: t_isD_16_1
122/159 Test #129: t_isD_16_1 .......................***Failed    0.20 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:26969... done.
new connection from 169.254.145.9:18403.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16718 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect
/mnt/i/DIY/CS144/sponge/txrx.sh: line 156: 16716 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"

        Start 130: t_isD_32K_d
123/159 Test #130: t_isD_32K_d ......................***Failed    0.17 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:15108... done.
new connection from 169.254.145.9:61200.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16737 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 131: t_icS_128K_8K_l
124/159 Test #131: t_icS_128K_8K_l ..................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:21014... done.
new connection from 169.254.144.9:57164.
DEBUG: Outbound stream to 169.254.144.9:57164 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.145.9:21014 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16758 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 132: t_icS_128K_8K_L
125/159 Test #132: t_icS_128K_8K_L ..................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:7364... done.
new connection from 169.254.144.9:316.
DEBUG: Outbound stream to 169.254.144.9:316 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.145.9:7364 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16778 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 133: t_icS_128K_8K_lL
126/159 Test #133: t_icS_128K_8K_lL .................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:13575... done.
new connection from 169.254.144.9:61776.
DEBUG: Outbound stream to 169.254.144.9:61776 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.145.9:13575 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16801 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 134: t_icR_128K_8K_l
127/159 Test #134: t_icR_128K_8K_l ..................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:3113... done.
DEBUG: Outbound stream to 169.254.145.9:3113 finished (1 byte still in flight).
new connection from 169.254.144.9:1182.
DEBUG: Inbound stream from 169.254.144.9:1182 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 135: t_icR_128K_8K_L
128/159 Test #135: t_icR_128K_8K_L ..................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:13715... done.
DEBUG: Outbound stream to 169.254.145.9:13715 finished (1 byte still in flight).
new connection from 169.254.144.9:28270.
DEBUG: Inbound stream from 169.254.144.9:28270 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 136: t_icR_128K_8K_lL
129/159 Test #136: t_icR_128K_8K_lL .................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:22838... done.
DEBUG: Outbound stream to 169.254.145.9:22838 finished (1 byte still in flight).
new connection from 169.254.144.9:29160.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 137: t_icD_128K_8K_l
130/159 Test #137: t_icD_128K_8K_l ..................***Failed    0.22 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:17599... done.
new connection from 169.254.144.9:8368.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16923 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16925 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 138: t_icD_128K_8K_L
131/159 Test #138: t_icD_128K_8K_L ..................***Failed    0.23 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:8369... done.
new connection from 169.254.144.9:2928.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16944 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16946 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 139: t_icD_128K_8K_lL
132/159 Test #139: t_icD_128K_8K_lL .................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.145.9:17115... done.
new connection from 169.254.144.9:4439.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16968 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 16970 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 140: t_isS_128K_8K_l
133/159 Test #140: t_isS_128K_8K_l ..................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:11572... done.
DEBUG: Outbound stream to 169.254.144.9:11572 finished (1 byte still in flight).
new connection from 169.254.145.9:41155.
DEBUG: Inbound stream from 169.254.145.9:41155 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 141: t_isS_128K_8K_L
134/159 Test #141: t_isS_128K_8K_L ..................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:13819... done.
DEBUG: Outbound stream to 169.254.144.9:13819 finished (1 byte still in flight).
new connection from 169.254.145.9:19311.
DEBUG: Inbound stream from 169.254.145.9:19311 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 142: t_isS_128K_8K_lL
135/159 Test #142: t_isS_128K_8K_lL .................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:5931... done.
DEBUG: Outbound stream to 169.254.144.9:5931 finished (1 byte still in flight).
new connection from 169.254.145.9:46156.
DEBUG: Inbound stream from 169.254.145.9:46156 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 143: t_isR_128K_8K_l
136/159 Test #143: t_isR_128K_8K_l ..................***Failed    0.21 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:16002... done.
new connection from 169.254.145.9:61929.
DEBUG: Outbound stream to 169.254.145.9:61929 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.9:16002 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17141 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 144: t_isR_128K_8K_L
137/159 Test #144: t_isR_128K_8K_L ..................***Failed    0.20 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:28589... done.
new connection from 169.254.145.9:51951.
DEBUG: Outbound stream to 169.254.145.9:51951 finished (1 byte still in flight).
DEBUG: Inbound stream from 169.254.144.9:28589 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17161 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 145: t_isR_128K_8K_lL
138/159 Test #145: t_isR_128K_8K_lL .................***Failed    0.21 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:22575... done.
new connection from 169.254.145.9:56820.
DEBUG: Outbound stream to 169.254.145.9:56820 finished (1 byte still in flight).
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17181 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 146: t_isD_128K_8K_l
139/159 Test #146: t_isD_128K_8K_l ..................***Failed    0.21 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:26406... done.
new connection from 169.254.145.9:361.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17204 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 147: t_isD_128K_8K_L
140/159 Test #147: t_isD_128K_8K_L ..................***Failed    0.21 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:20902... done.
new connection from 169.254.145.9:60603.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17225 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17227 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 148: t_isD_128K_8K_lL
141/159 Test #148: t_isD_128K_8K_lL .................***Failed    0.19 sec
DEBUG: Listening for incoming connection... DEBUG: Connecting to 169.254.144.9:9124... done.
new connection from 169.254.145.9:18754.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
  what():  std::exceptionterminate called after throwing an instance of 'std::exception'

  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17246 Aborted                 coproc COPROC $3 -l $4 ${SERVER_PORT} > "$1" < "$2"
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17248 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 149: t_icnS_128K_8K_l
142/159 Test #149: t_icnS_128K_8K_l .................***Failed    0.26 sec
DEBUG: Connecting to 169.254.144.1:18486... done.
DEBUG: Inbound stream from 169.254.144.1:18486 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17268 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 150: t_icnS_128K_8K_L
143/159 Test #150: t_icnS_128K_8K_L .................***Failed    0.24 sec
DEBUG: Connecting to 169.254.144.1:10426... done.
DEBUG: Inbound stream from 169.254.144.1:10426 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17291 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 151: t_icnS_128K_8K_lL
144/159 Test #151: t_icnS_128K_8K_lL ................***Timeout  10.00 sec
DEBUG: Connecting to 169.254.144.1:25707... 

        Start 152: t_icnR_128K_8K_l
145/159 Test #152: t_icnR_128K_8K_l .................***Timeout  10.00 sec
DEBUG: Connecting to 169.254.144.1:22539... done.
DEBUG: Outbound stream to 169.254.144.1:22539 finished (1 byte still in flight).
DEBUG: Outbound stream to 169.254.144.1:22539 has been fully acknowledged.

        Start 153: t_icnR_128K_8K_L
146/159 Test #153: t_icnR_128K_8K_L .................***Timeout  10.00 sec
DEBUG: Connecting to 169.254.144.1:31898... done.
DEBUG: Outbound stream to 169.254.144.1:31898 finished (1 byte still in flight).
DEBUG: Outbound stream to 169.254.144.1:31898 has been fully acknowledged.

        Start 154: t_icnR_128K_8K_lL
147/159 Test #154: t_icnR_128K_8K_lL ................***Timeout  10.00 sec
DEBUG: Connecting to 169.254.144.1:26323... done.
DEBUG: Outbound stream to 169.254.144.1:26323 finished (1 byte still in flight).
DEBUG: Outbound stream to 169.254.144.1:26323 has been fully acknowledged.

        Start 155: t_icnD_128K_8K_l
148/159 Test #155: t_icnD_128K_8K_l .................***Failed    0.24 sec
DEBUG: Connecting to 169.254.144.1:26321... done.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17491 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 156: t_icnD_128K_8K_L
149/159 Test #156: t_icnD_128K_8K_L .................***Failed    0.26 sec
DEBUG: Connecting to 169.254.144.1:3290... done.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17514 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 157: t_icnD_128K_8K_lL
150/159 Test #157: t_icnD_128K_8K_lL ................***Failed    0.27 sec
DEBUG: Connecting to 169.254.144.1:15867... done.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception
/mnt/i/DIY/CS144/sponge/txrx.sh: line 110: 17534 Aborted                 $3 $4 ${SERVER_PORT} > "$1" < "$2"
Error in _rt_connect

        Start 158: t_isnS_128K_8K_l
151/159 Test #158: t_isnS_128K_8K_l .................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... new connection from 169.254.144.1:39008.
DEBUG: Inbound stream from 169.254.144.1:39008 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 159: t_isnS_128K_8K_L
152/159 Test #159: t_isnS_128K_8K_L .................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... new connection from 169.254.144.1:48536.
DEBUG: Inbound stream from 169.254.144.1:48536 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 160: t_isnS_128K_8K_lL
153/159 Test #160: t_isnS_128K_8K_lL ................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... new connection from 169.254.144.1:36996.
DEBUG: Inbound stream from 169.254.144.1:36996 finished cleanly.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 161: t_isnR_128K_8K_l
154/159 Test #161: t_isnR_128K_8K_l .................***Timeout  10.00 sec
DEBUG: Listening for incoming connection... new connection from 169.254.144.1:40164.
DEBUG: Outbound stream to 169.254.144.1:40164 finished (1 byte still in flight).
DEBUG: Outbound stream to 169.254.144.1:40164 has been fully acknowledged.

        Start 162: t_isnR_128K_8K_L
155/159 Test #162: t_isnR_128K_8K_L .................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... new connection from 169.254.144.1:56822.
DEBUG: Outbound stream to 169.254.144.1:56822 finished (1 byte still in flight).
DEBUG: Outbound stream to 169.254.144.1:56822 has been fully acknowledged.

        Start 163: t_isnR_128K_8K_lL
156/159 Test #163: t_isnR_128K_8K_lL ................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... new connection from 169.254.144.1:48214.
DEBUG: Outbound stream to 169.254.144.1:48214 finished (1 byte still in flight).
DEBUG: Outbound stream to 169.254.144.1:48214 has been fully acknowledged.

        Start 164: t_isnD_128K_8K_l
157/159 Test #164: t_isnD_128K_8K_l .................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... new connection from 169.254.144.1:44680.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 165: t_isnD_128K_8K_L
158/159 Test #165: t_isnD_128K_8K_L .................***Timeout  10.01 sec
DEBUG: Listening for incoming connection... new connection from 169.254.144.1:49678.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception

        Start 166: t_isnD_128K_8K_lL
159/159 Test #166: t_isnD_128K_8K_lL ................***Timeout  10.03 sec
DEBUG: Listening for incoming connection... new connection from 169.254.144.1:38750.
Exception in TCPConnection runner thread: TCPConnection::write() accepted less than advertised length
terminate called after throwing an instance of 'std::exception'
  what():  std::exception


22% tests passed, 124 tests failed out of 159

Total Test time (real) = 550.70 sec

The following tests FAILED:
	 33 - t_active_close (Failed)
	 34 - t_passive_close (Failed)
	 36 - t_ack_rst (Failed)
	 38 - t_ack_rst_win (Failed)
	 43 - t_winsize (Failed)
	 45 - t_retx (Failed)
	 46 - t_retx_win (Failed)
	 47 - t_loopback (Failed)
	 48 - t_loopback_win (Failed)
	 49 - t_reorder (Failed)
	 53 - t_udp_client_send (Failed)
	 54 - t_udp_server_send (Timeout)
	 55 - t_udp_client_recv (Timeout)
	 56 - t_udp_server_recv (Failed)
	 57 - t_udp_client_dupl (Failed)
	 58 - t_udp_server_dupl (Failed)
	 59 - t_ucS_1M_32k (Failed)
	 60 - t_ucS_128K_8K (Failed)
	 61 - t_ucS_16_1 (Failed)
	 62 - t_ucS_32K_d (Failed)
	 63 - t_ucR_1M_32k (Timeout)
	 64 - t_ucR_128K_8K (Timeout)
	 65 - t_ucR_16_1 (Timeout)
	 66 - t_ucR_32K_d (Timeout)
	 67 - t_ucD_1M_32k (Failed)
	 68 - t_ucD_128K_8K (Failed)
	 69 - t_ucD_16_1 (Failed)
	 70 - t_ucD_32K_d (Failed)
	 71 - t_usS_1M_32k (Timeout)
	 72 - t_usS_128K_8K (Timeout)
	 73 - t_usS_16_1 (Timeout)
	 74 - t_usS_32K_d (Timeout)
	 75 - t_usR_1M_32k (Failed)
	 76 - t_usR_128K_8K (Failed)
	 77 - t_usR_16_1 (Failed)
	 78 - t_usR_32K_d (Failed)
	 79 - t_usD_1M_32k (Failed)
	 80 - t_usD_128K_8K (Failed)
	 81 - t_usD_16_1 (Failed)
	 82 - t_usD_32K_d (Failed)
	 83 - t_ucS_128K_8K_l (Failed)
	 84 - t_ucS_128K_8K_L (Failed)
	 85 - t_ucS_128K_8K_lL (Timeout)
	 86 - t_ucR_128K_8K_l (Timeout)
	 87 - t_ucR_128K_8K_L (Timeout)
	 88 - t_ucR_128K_8K_lL (Timeout)
	 89 - t_ucD_128K_8K_l (Timeout)
	 90 - t_ucD_128K_8K_L (Failed)
	 91 - t_ucD_128K_8K_lL (Failed)
	 92 - t_usS_128K_8K_l (Timeout)
	 93 - t_usS_128K_8K_L (Timeout)
	 94 - t_usS_128K_8K_lL (Timeout)
	 95 - t_usR_128K_8K_l (Timeout)
	 96 - t_usR_128K_8K_L (Timeout)
	 97 - t_usR_128K_8K_lL (Failed)
	 98 - t_usD_128K_8K_l (Failed)
	 99 - t_usD_128K_8K_L (Failed)
	100 - t_usD_128K_8K_lL (Failed)
	101 - t_ipv4_client_send (Failed)
	102 - t_ipv4_server_send (Timeout)
	103 - t_ipv4_client_recv (Timeout)
	104 - t_ipv4_server_recv (Failed)
	105 - t_ipv4_client_dupl (Failed)
	106 - t_ipv4_server_dupl (Failed)
	107 - t_icS_1M_32k (Failed)
	108 - t_icS_128K_8K (Failed)
	109 - t_icS_16_1 (Failed)
	110 - t_icS_32K_d (Failed)
	111 - t_icR_1M_32k (Timeout)
	112 - t_icR_128K_8K (Timeout)
	113 - t_icR_16_1 (Timeout)
	114 - t_icR_32K_d (Timeout)
	115 - t_icD_1M_32k (Failed)
	116 - t_icD_128K_8K (Failed)
	117 - t_icD_16_1 (Failed)
	118 - t_icD_32K_d (Failed)
	119 - t_isS_1M_32k (Timeout)
	120 - t_isS_128K_8K (Timeout)
	121 - t_isS_16_1 (Timeout)
	122 - t_isS_32K_d (Timeout)
	123 - t_isR_1M_32k (Failed)
	124 - t_isR_128K_8K (Failed)
	125 - t_isR_16_1 (Failed)
	126 - t_isR_32K_d (Failed)
	127 - t_isD_1M_32k (Failed)
	128 - t_isD_128K_8K (Failed)
	129 - t_isD_16_1 (Failed)
	130 - t_isD_32K_d (Failed)
	131 - t_icS_128K_8K_l (Failed)
	132 - t_icS_128K_8K_L (Failed)
	133 - t_icS_128K_8K_lL (Failed)
	134 - t_icR_128K_8K_l (Timeout)
	135 - t_icR_128K_8K_L (Timeout)
	136 - t_icR_128K_8K_lL (Timeout)
	137 - t_icD_128K_8K_l (Failed)
	138 - t_icD_128K_8K_L (Failed)
	139 - t_icD_128K_8K_lL (Failed)
	140 - t_isS_128K_8K_l (Timeout)
	141 - t_isS_128K_8K_L (Timeout)
	142 - t_isS_128K_8K_lL (Timeout)
	143 - t_isR_128K_8K_l (Failed)
	144 - t_isR_128K_8K_L (Failed)
	145 - t_isR_128K_8K_lL (Failed)
	146 - t_isD_128K_8K_l (Failed)
	147 - t_isD_128K_8K_L (Failed)
	148 - t_isD_128K_8K_lL (Failed)
	149 - t_icnS_128K_8K_l (Failed)
	150 - t_icnS_128K_8K_L (Failed)
	151 - t_icnS_128K_8K_lL (Timeout)
	152 - t_icnR_128K_8K_l (Timeout)
	153 - t_icnR_128K_8K_L (Timeout)
	154 - t_icnR_128K_8K_lL (Timeout)
	155 - t_icnD_128K_8K_l (Failed)
	156 - t_icnD_128K_8K_L (Failed)
	157 - t_icnD_128K_8K_lL (Failed)
	158 - t_isnS_128K_8K_l (Timeout)
	159 - t_isnS_128K_8K_L (Timeout)
	160 - t_isnS_128K_8K_lL (Timeout)
	161 - t_isnR_128K_8K_l (Timeout)
	162 - t_isnR_128K_8K_L (Timeout)
	163 - t_isnR_128K_8K_lL (Timeout)
	164 - t_isnD_128K_8K_l (Timeout)
	165 - t_isnD_128K_8K_L (Timeout)
	166 - t_isnD_128K_8K_lL (Timeout)
Errors while running CTest
CMakeFiles/check_lab4.dir/build.make:57: recipe for target 'CMakeFiles/check_lab4' failed
make[3]: *** [CMakeFiles/check_lab4] Error 8
CMakeFiles/Makefile2:131: recipe for target 'CMakeFiles/check_lab4.dir/all' failed
make[2]: *** [CMakeFiles/check_lab4.dir/all] Error 2
CMakeFiles/Makefile2:138: recipe for target 'CMakeFiles/check_lab4.dir/rule' failed
make[1]: *** [CMakeFiles/check_lab4.dir/rule] Error 2
Makefile:155: recipe for target 'check_lab4' failed
make: *** [check_lab4] Error 2
